### PR TITLE
[IMP] product,sale(_pdf_quote_builder): embed further pdf quote builder features

### PR DESCRIPTION
**Pull request by vava-odoo**
This commit improves the user experience with pdf quote builder by
- showing document for both product template and product variants
- providing pdf example for headers/footers

task-3543479

### CODE REVIEW

File: addons/sale_pdf_quote_builder/wizards/res_config_settings_views.xml
Comment (Feyensv) [inline addons/sale_pdf_quote_builder/wizards/res_config_settings_views.xml:22]: IMHO, you won't be able to make it work unless you make a custom js widget for that (or improve the standard one for binary fields).

File: addons/product/views/product_document_views.xml
Comment (Demesmaeker) [inline addons/product/views/product_document_views.xml:23]:
```suggestion
                        <field name="res_name" string="Product Variant" invisible="res_model != 'product.product'"/>
```
Might be exactly the same, but would be more logical IMHO, this is only a suggestion though, feel free to ignore

File: addons/product/views/product_document_views.xml
Comment (Demesmaeker) [inline addons/product/views/product_document_views.xml:67]: Same here. It slightly disturbs that to set something as a variant, you are looking if it's a template, instead of checking if it's a variant. Personal preference though.

**Review by Demesmaeker — APPROVED** [review]
Hello :duck: 

IMHO, these changes should be in stable. The view and data changes are borderline, they shouldn't break anything, but... =/
Each module changes are independent enough that even without updating all the modules at the same time, I don't see any problem in doing it, but...
Changes to the fields won't be applied until the server is restarted, resulting in slightly incoherent but otherwise non-breaking data, but...
This is the stable version, this isn't "fixing" anything, it's improvement and shouldn't be done in stable, as this is against the stable policy.

I won't block this PR, as I know it's a request and that it won't break anything for existing DB. So I'll green that task anyway, seeing as the comments are only minor changes, and let the final reviewer take the final decision.

File: addons/product/models/product_product.py
Comment (Feyensv) [inline addons/product/models/product_product.py:67]: So if I have a document on `product.product` 3, it will be shown in the documents of `product.template` 3 as well ? :sparkles:

Comment (vava-odoo) [reply addons/product/models/product_product.py]: Woopsie

File: addons/product/models/product_template.py
Comment (Feyensv) [inline addons/product/models/product_template.py:174]: Same problem

File: addons/product/models/product_template.py
Comment (Feyensv) [inline addons/product/models/product_template.py:610]: Same problem

File: addons/product/models/product_product.py
Comment (Demesmaeker) [reply addons/product/models/product_product.py]: Well... Yes, but isn't that the whole point of this task? From what I understand... But the variants are shown with a ruban... So yes, that's what's wanted... oO Did I read the task wrong? xD

Comment (Feyensv) [reply addons/product/models/product_product.py]: There should be no modification on the x2many field, it's not shown anywhere.
But the actions should be adapted to a domain:
```
('res_model', '=', 'product.template'), ('res_id', '=', template.id),
OR
('res_model', '=', 'product.product'), ('res_id', 'in', variants.ids),
```

File: addons/product/models/product_template.py
Comment (Feyensv) [inline addons/product/models/product_template.py:620]: This domain should be replaced when `action_open_documents` is called from the `product.product`, the domain should only include the template document or the variant ones imho, not all the documents of the template variants :thinking: 
image

Comment (vava-odoo) [reply addons/product/models/product_template.py:620]: That needs to be discussed with @dalaOdoo. From what I recall, the idea is that the filter `search_default_context_variant` is applied first to restrict to the current variant, but if you remove the filter you'd like a global view on all products from the related template.

Comment (Feyensv) [reply addons/product/models/product_template.py:620]: Yeah, I don't strictly mind, but I wanted to be sure that's really the expected behavior :ok_hand:

Comment (Feyensv) [reply addons/product/models/product_template.py:620]: If you remove the "Documents of this variant" filter when coming from a variant, what do you expect to see ?
Documents of the template and of the variant you are coming from or the documents of all variants of the template ?

Comment (dalaOdoo) [reply addons/product/models/product_template.py:620]: Theoretically, only the documents that matters for this variant so the ones of the template (no of other variants) but practically I feel like displaying everything may help the user : to duplicate, to find a document wrongly configured, ...

Comment (Feyensv) [reply addons/product/models/product_template.py:620]: Mhm, ok then, let's keep it like this.

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
